jax/_src/scipy/special.py

```diff
@@ -191,6 +190,47 @@ def multigammaln(a: ArrayLike, d: ArrayLike) -> Array:
   return res + constant
 
 
+@_wraps(osp_special.kl_div, module="scipy.special")
+def kl_div(
+    p: ArrayLike,
+    q: ArrayLike,
+) -> Array:
+    p, q = promote_args_inexact("kl_div", p, q)
+    both_gt_zero_mask = lax.bitwise_and(lax.gt(p, 0.0), lax.gt(q, 0.0))
+    one_zero_mask = lax.bitwise_and(lax.eq(p, 0.0), lax.ge(q, 0.0))
```

Comment:
This is failing with a dtype mismatch with float32 inputs and `JAX_ENABLE_X64=1`.

If you want to use `lax` functions here, the input types must match exactly. You could use `zero = _lax_const(p, 0.0)` to ensure this is the case.